sqlalchemy_ibmi/base.py

```diff
@@ -261,102 +302,103 @@ def visit_VARGRAPHIC(self, type_):
-        elif not type_.scale:
+        if not type_.scale:
             return "DECIMAL(%(precision)s, 0)" % {'precision': type_.precision}
-        else:
-            return "DECIMAL(%(precision)s, %(scale)s)" % {
-                            'precision': type_.precision, 'scale': type_.scale}
+        return "DECIMAL(%(precision)s, %(scale)s)" % {
+            'precision': type_.precision, 'scale': type_.scale}
 
-
-    def visit_numeric(self, type_):
+    def visit_numeric(self, type_, **kw):
         return self.visit_DECIMAL(type_)
 
-    def visit_datetime(self, type_):
+    def visit_datetime(self, type_, **kw):
         return self.visit_TIMESTAMP(type_)
 
-    def visit_date(self, type_):
+    def visit_date(self, type_, **kw):
         return self.visit_DATE(type_)
 
-    def visit_time(self, type_):
+    def visit_time(self, type_, **kw):
         return self.visit_TIME(type_)
 
-    def visit_integer(self, type_):
+    def visit_integer(self, type_, **kw):
         return self.visit_INT(type_)
 
-    def visit_boolean(self, type_):
+    def visit_boolean(self, type_, **kw):
         return self.visit_SMALLINT(type_)
 
-    def visit_float(self, type_):
+    def visit_float(self, type_, **kw):
         return self.visit_FLOAT(type_)
 
-    def visit_unicode(self, type_):
+    def visit_unicode(self, type_, **kw):
         return self.visit_VARGRAPHIC(type_)
 
-    def visit_unicode_text(self, type_):
+    def visit_unicode_text(self, type_, **kw):
         return self.visit_LONGVARGRAPHIC(type_)
 
-    def visit_string(self, type_):
+    def visit_string(self, type_, **kw):
         return self.visit_VARCHAR(type_)
 
-    def visit_TEXT(self, type_):
+    def visit_TEXT(self, type_, **kw):
         return self.visit_CLOB(type_)
 
-    def visit_large_binary(self, type_):
+    def visit_large_binary(self, type_, **kw):
         return self.visit_BLOB(type_)
 
 
 class DB2Compiler(compiler.SQLCompiler):
+    """IBM i DB2 compiler class"""
+
+    def visit_empty_set_expr(self, element_types):
+        pass
+
+    def update_from_clause(self, update_stmt, from_table, extra_froms,
+                           from_hints, **kw):
+        pass
+
+    def delete_extra_from_clause(self, update_stmt, from_table, extra_froms,
+                                 from_hints, **kw):
+        pass
 
-    if SA_Version < [0, 9]:
-        def visit_false(self, expr, **kw):
-            return '0'
-            
-        def visit_true(self, expr, **kw):
-            return '1'
-            
     def get_cte_preamble(self, recursive):
         return "WITH"
-        
+
     def visit_now_func(self, fn, **kw):
         return "CURRENT_TIMESTAMP"
 
-    def for_update_clause(self, select):
-        if select.for_update == True:
+    def for_update_clause(self, select, **kw):
+        if select.for_update:
             return ' WITH RS USE AND KEEP UPDATE LOCKS'
-        elif select.for_update == 'read':
+        if select.for_update == 'read':
```

Comment:
As written, the second if will never happen since 'read' is _truthy_ will be selected by the first if condition. These conditions should be switched around.